plugins/query-resource-limit-group/src/main/java/org/opensearch/plugin/resource_limit_group/CreateResourceLimitGroupRequest.java

```diff
@@ -0,0 +1,182 @@
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.resource_limit_group;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.cluster.metadata.ResourceLimitGroup;
+import org.opensearch.cluster.metadata.ResourceLimitGroup.ResourceLimit;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * A request for create Resource Limit Group
+ *
+ * @opensearch.internal
+ */
+public class CreateResourceLimitGroupRequest extends ActionRequest implements Writeable.Reader<CreateResourceLimitGroupRequest> {
+    private String name;
+    private String uuid;
+    private List<ResourceLimit> resourceLimits;
+    private String enforcement;
+    private String createdAt;
+    private String updatedAt;
+
+    /**
+     * Default constructor for CreateResourceLimitGroupRequest
+     */
+    public CreateResourceLimitGroupRequest() {}
+
+    /**
+     * Constructor for CreateResourceLimitGroupRequest
+     * @param resourceLimitGroup - A {@link ResourceLimitGroup} object
+     */
+    public CreateResourceLimitGroupRequest(ResourceLimitGroup resourceLimitGroup) {
+        this.name = resourceLimitGroup.getName();
+        this.uuid = resourceLimitGroup.getUUID();
+        this.resourceLimits = resourceLimitGroup.getResourceLimits();
+        this.enforcement = resourceLimitGroup.getEnforcement();
+        this.createdAt = resourceLimitGroup.getCreatedAt();
+        this.updatedAt = resourceLimitGroup.getUpdatedAt();
+    }
+
+    /**
+     * Constructor for CreateResourceLimitGroupRequest
+     * @param in - A {@link StreamInput} object
+     */
+    public CreateResourceLimitGroupRequest(StreamInput in) throws IOException {
+        super(in);
+        name = in.readString();
+        uuid = in.readString();
+        resourceLimits = in.readList(ResourceLimit::new);
+        enforcement = in.readString();
+        createdAt = in.readString();
+        updatedAt = in.readString();
+    }
+
+    @Override
+    public CreateResourceLimitGroupRequest read(StreamInput in) throws IOException {
+        return new CreateResourceLimitGroupRequest(in);
+    }
+
+    /**
+     * Generate a CreateResourceLimitGroupRequest from XContent
+     * @param parser - A {@link XContentParser} object
+     */
+    public static CreateResourceLimitGroupRequest fromXContent(XContentParser parser) throws IOException {
+        ResourceLimitGroup resourceLimitGroup = ResourceLimitGroup.fromXContent(parser);
+        return new CreateResourceLimitGroupRequest(resourceLimitGroup);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
```

Comment:
Why are we not validating the request ?

Reply:
When a request enters the rest class (eg. RestUpdateResourceLimitGroupAction), we call the function  UpdateResourceLimitGroupRequest.fromXContent(parser); to parse the new request. This function will call ResourceLimitGroup.fromXContentOptionalFields/ResourceLimitGroup.fromXContent, and within witch will have the validator for each resource limit group. As a result, all fields in UpdateResourceLimitGroupRequest already passed the validator in ResourceLimitGroup. Thus this validate() function can safely return null here.